hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -401,6 +406,50 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 			log.Info("hostedcluster is still deleting", "name", req.NamespacedName)
 			return ctrl.Result{RequeueAfter: clusterDeletionRequeueDuration}, nil
 		}
+
+		// This new condition is necessary for OCM people to report the cloud dangling objects to the user.
+		// The grace period is customizable using the flag --destroy-grace-period it's a time.Duration field type.
+		// before remove the finalizer on the HostedCluster. This condition only will be created when !SelfManage and
+		// CloudResourcesDestroyedCondition is False.
+		CloudResourcesDestroyedCondition := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.CloudResourcesDestroyed))
+		if CloudResourcesDestroyedCondition.Status == "False" {
+			// ToDo (jparrill): when you deploy/destroy the cluster form CLI, you need to known the HCDestroyGracePeriod if not, on destroy
```

Comment:
Could you create a Jira ticket for this and note the ticket # here?

Reply:
https://issues.redhat.com/browse/HOSTEDCP-1154